### DISCUSSION TITLE
Config option to send payloads to a logger when Ray is disabled

### DISCUSSION THREAD

**Discussion by squatto** [spatie/laravel-ray#113] [Ideas]
My idea will make more sense with an example for context:

```php
if ($response->failed()) {
    if (ray()->enabled()) {
        ray('Response JSON:', $response->json());
    } else {
        info('Response JSON:');
        info(json_encode($response->json(), JSON_PRETTY_PRINT));
    }
}
```

`$response` is the return from the Laravel HTTP client. When the request fails, I send the JSON through `ray()` if Ray is enabled (e.g. because it's on `local`). If Ray is disabled (e.g. because it's on `production`), I still want to log the JSON response, so I send it through `info()`. This `if (ray()->enabled()) { ... } else { ... }` process lets me keep `ray()` calls in code pushed to production (where Ray is disabled) and continue receiving log output whenever necessary.

It would be very helpful to be able to _only_ do the `ray()` call, and if Ray is disabled, have the option to pass the payload output to a logger that is configured in `ray.php` (any PSR-3 compliant logger)

This would likely necessitate some payloads having a "simplified" formatter - one whose output could be easily saved to a text-based log file, for example. For most of the payloads this would be pretty trivial to accomplish. I'd imagine that most (if not all) payloads could just send `$payload->toArray()` to the relevant log method. If necessary, the payloads could override a method from `\Spatie\Ray\Payloads\Payload` (like they do now with `getContent()`) if a different payload format is necessary/desired. Something like `getContentForLogger()` that by default just returns `$this->toArray()`, maybe?

```php
public function getContentForLogger(): array
{
    return $this->toArray();
}
```

Thanks!

**Comment by patinthehat** [comment]
@squatto Couldn't you do something like this and override the `spatie/ray` `ray()` function? _(code is untested, but you get the idea)_

```php
function ray(...$args) {
    $settings = SettingsFactory::createFromConfigFile();

    if (Ray::$enabled) {        
        return (new Ray($settings))->send(...$args);
    }

    info($args);
    return new Ray($settings);
}

ray($response->json());
```
You could also create a custom Ray class and override the `send()` method.  
Just my thoughts on how you could solve your issue.
